scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2878,252 @@ cdef class Rotation:
                              ", got {} and {} respectively.".format(
                                 a.shape, b.shape))
 
+        # Find number of vector pairs, and normalize to unit length
+        if a.shape == (3,):
+            N = 1
+            _normalize3(a)
+            _normalize3(b)
+            a3 = a
+            b3 = b
+        else:
+            N = len(a)
+            a /= np.linalg.norm(a, axis=1).reshape(N, 1)
+            b /= np.linalg.norm(b, axis=1).reshape(N, 1)
+            if N == 1:
+                a3 = a.reshape(3)
+                b3 = b.reshape(3)
+
+        # Check weights
         if weights is None:
-            weights = np.ones(len(b))
+            weights = np.ones(N)
         else:
             weights = np.asarray(weights)
-            if weights.ndim != 1:
-                raise ValueError("Expected `weights` to be 1 dimensional, got "
-                                 "shape {}.".format(weights.shape))
-            if weights.shape[0] != b.shape[0]:
-                raise ValueError("Expected `weights` to have number of values "
-                                 "equal to number of input vectors, got "
-                                 "{} values and {} vectors.".format(
-                                    weights.shape[0], b.shape[0]))
-            if (weights < 0).any():
-                raise ValueError("`weights` may not contain negative values")
+        if weights.ndim != 1:
+            raise ValueError("Expected `weights` to be 1 dimensional, got "
+                             "shape {}.".format(weights.shape))
+        if N > 1 and (weights.shape[0] != b.shape[0]):
+            raise ValueError("Expected `weights` to have number of values"
+                             "equal to number of input vectors, got "
+                             "{} values and {} vectors.".format(
+                                weights.shape[0], b.shape[0]))
+        if (weights < 0).any():
+            raise ValueError("`weights` may not contain negative values")
+
+        # Check for an infinite weight, which indicates that the corresponding
+        # vector pair is the primary unmoving reference which we align the
+        # other vectors to.
+        weight_inf = np.isposinf(weights)
+        if sum(weight_inf) > 1:
+            raise ValueError("Only one infinite weight is allowed")
+        elif N >= 2 and weight_inf.any():
+            if N == 2:
+                # A single non-infinite weight vector only needs normalization
+                a_non_inf = a[~weight_inf].reshape(3)
+                b_non_inf = b[~weight_inf].reshape(3)
+            else:  # N > 2
+                # Multiple non-infinite weight vectors need to be best aligned
+                R, _ = cls.align_vectors(a[~weight_inf], b[~weight_inf],
+                                         weights=weights[~weight_inf])
+                R_mat = R.as_matrix()
+                a_non_inf = R_mat[0, :]  # Already normalized
```

Comment:
Hmm, I can see why it can work, but do you have any mathematical details why it corresponds to the original alignment problem with constraints?

Reply:
I'm not quite sure this is working correctly... need to think about it.

Reply:
Yes this is definitely incorrect right now, figuring out how to fix it.

Reply:
My sort of intuitive issues with that was: why take axis 0 from `R_mat` here, why not axis 1 or 2?